cpp/contracts/feesmanager.cpp

```diff
@@ -0,0 +1,86 @@
+#include "feesmanager.hpp"
+
+using namespace eosio;
+
+void feesmanager::setallowance( name node, name token, const asset& value ) {
+    require_auth(get_self());
+
+    auto balance = getbalance(token, value.symbol);
+
+    total_allowance_table total_allowance(get_self(), token.value);
+    auto total_token_allowance = total_allowance.get_or_create(get_self(), {asset(0, value.symbol)}).allowance.amount;
+
+    allowances_table allowances(get_self(), node.value);
+    auto allowance_itr = allowances.find(value.symbol.code().raw());
+    auto allowance_name_idx = allowances.get_index<"name"_n>();
+    auto allowance_name_itr = allowance_name_idx.find(token.value);
+
+    if (allowance_itr == allowances.end() && allowance_name_itr == allowance_name_idx.end()) {
+        check(balance.amount >= total_token_allowance + value.amount, "balance is lower than the allowance to be set");
+        allowances.emplace(get_self(), [&](auto& a) {
+            a.node_allowance = value;
+            a.token = token;
+        });
+        total_allowance.set({asset(total_token_allowance + value.amount , value.symbol)}, get_self());
+    } else {
+        check(allowance_itr->token == token, "symbol and token do not match");
+        check(balance.amount >= (total_token_allowance - allowance_itr->node_allowance.amount) + value.amount, "balance is lower than the allowance to be set");
```

Comment:
I'd differentiate this check msg from the previous one, so we can know on which branch of the if we are